eth-providers/src/utils/transactionHelper.ts

```diff
@@ -0,0 +1,69 @@
+import { BigNumber, BigNumberish } from 'ethers';
+
+type TxConsts = {
+  storageByteDeposit: BigNumberish;
+  txFeePerGas: BigNumberish;
+};
+
+const bigNumDiv = (x: BigNumber, y: BigNumber) => {
+  const res = x.div(y);
+  return res.mul(y) === x ? res : res.add(1);
+};
+
+export const calcEthereumTransactionParams = (
+  data: {
+    gasLimit: BigNumberish;
+    storageLimit: BigNumberish;
+    validUntil: BigNumberish;
+  } & TxConsts
+): {
+  txGasPrice: BigNumber;
+  txGasLimit: BigNumber;
+} => {
+  const gasLimit = BigNumber.from(data.gasLimit);
+  const storageLimit = BigNumber.from(data.storageLimit);
+  const validUntil = BigNumber.from(data.validUntil);
+  const storageByteDeposit = BigNumber.from(data.storageByteDeposit);
+  const txFeePerGas = BigNumber.from(data.txFeePerGas);
+
+  const blockPeriod = validUntil.div(30);
+  const storageEntryLimit = storageLimit.div(64);
```

Comment:
Should use `bigNumDiv` here.

Reply:
if we use `bigNumDivRoundUp` as the method name, this might be more clear haha